src/cider/nrepl/middleware/test.clj

```diff
@@ -227,6 +259,20 @@
       (report {:type :error, :fault true, :expected nil, :actual e
                :message "Uncaught exception in test fixture"}))))
 
+(defmacro ^:private timing
+  "Executes `body`, reporting the time it took by persisting it to `time-atom`."
+  {:style/indent 1}
+  [time-atom & body]
+  {:pre [(seq body)]}
+  `(let [then# (System/currentTimeMillis)
+         v# (do
+              ~@body)
+         took# (- (System/currentTimeMillis)
+                  then#)]
+     (reset! ~time-atom {:took-ms took#
```

Comment:
I'd probably use names like `:elapsed-time` and `:elapsed-time-string/humanized`, as I find the use of `took` confusing. Other terms that come to mind duration, time, etc. I recall `prepl` just uses the `:ms` key, so I can live with something like this as well.